gpflow/quadrature.py

```diff
@@ -77,3 +78,59 @@ def mvnquad(func, means, covs, H, Din=None, Dout=None):
     wr = np.reshape(wn * np.pi ** (-Din * 0.5),
                     (-1,) + (1,) * (1 + len(Dout)))
     return tf.reduce_sum(fX * wr, 0)
+
+
+def ndiagquad(funcs, H, Fmu, Fvar, **Ys):
```

Comment:
I'm happy to change the name - I just based it on mvnquad (but instead of multivariate normal it's a diagonal normal) - please suggest something better!

I think it makes more sense to have the integrand up front. funcs can be a single value (in which case the return value is a single tensor) or a tuple/list (in which case the return value is a tensor of the same length). Makes sense to me...

Reply:
Re types, as Union[tf.Tensor, np.ndarray] is going to crop up all over the place I think it'd be good to put that in a single place in GPflow instead of redefining it everywhere.